### PR TITLE
GN-4706: Table cell background

### LINKED ISSUE #1161: Background color cell
<p>Allow the user to set the bg color of a cell. SHould be inline styles, working kind of like the text marking feature we have but specifically for cells. Use a similar color selection menu for consistency. Add to both the toolbar table menu and the floating menu.</p>

### PR DESCRIPTION
**Pull request by dkozickis**
##### connected issues and PRs:

Closes #1161 

### Setup

1. Checkout

### How to test/reproduce

Repeat after me:

https://github.com/lblod/ember-rdfa-editor/assets/769698/50bd46eb-4b5c-41b4-8219-2f059a93e457

### Challenges/uncertainties

Design 😅 

### Checks PR readiness
- [x] UI: works on smaller screen sizes
- [x] UI: feedback for any loading/error states
- [x] Check cancel/go-back flows

Uploading CleanShot 2024-03-14 at 12.21.26.mp4…

- [x] Check database state correct when deleting/updating (especially regarding relationships)
- [x] changelog
- [x] npm lint
- [x] no new deprecations

### CODE REVIEW

**Review by piemonkey — APPROVED** [review]
Looks good. I feel like having both the toolbar button and the color choice in the (very aggressive in my view) tooltip menu seems like too much, but maybe that's just me.

File: addon/components/plugins/table/color.hbs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{{#if @controller}}
+  <div class="say-toolbar__button say-toolbar__highlight-button">
+    <this.Velcro @placement="bottom" @offsetOptions={{hash mainAxis=6}} as |velcro|>
+      <button
+        type='button'
+        class='main {{if @active "is-active"}}'
+        disabled={{not this.canSetColor}}
+        title={{t 'ember-rdfa-editor.color.button-label'}}
```
Comment (elpoelma) [inline addon/components/plugins/table/color.hbs:8]: We should use another label for this.

File: addon/components/plugins/table/color.ts
Comment (elpoelma) [inline addon/components/plugins/table/color.ts:1]: I'm wondering if we should include a seperate dropdown button, like we use for the highlight toolbar button.
This way the user can use between quickly applying a color and selecting a different color.
Not sure what the best approach would be here.

File: addon/plugins/table/nodes/table.ts [hunk 2]
```diff
@@ -106,24 +110,78 @@ interface TableNodes extends Record<string, NodeSpec> {
...
+  borderLeft: {
+    default: inlineBorderStyle ?? null,
+    getFromDOM(dom) {
+      return dom.style.borderLeft || null;
+    },
+    setDOMAttr(value, attrs) {
+      if (typeof value === 'string') {
+        appendToStyleAttribute(attrs, `border-left: ${value}`);
+      }
+    },
```
Comment (elpoelma) [inline addon/plugins/table/nodes/table.ts:155]: Nice approach to include seperate `fromDOM` and `toDOM` methods for each of the attributes. I was already thinking of including that option in the default `SayNodeSpec` API.

**Review by elpoelma — CHANGES_REQUESTED** [review]
Works well overall, just some remarks.

- We should (as you mentioned) review the design.
- We should probably also use a different icon to avoid confusion.
- I'm wondering if it is possible to include the color picker in the floating tooltip?

**Comment by abeforgit** [comment]
> Looks good. I feel like having both the toolbar button and the color choice in the (very aggressive in my view) tooltip menu seems like too much, but maybe that's just me.

having both is indeed much, I think in apps we'd have either/or though
for the tooltip menu I agree its a bit in your face atm. You have ideas on how to improve it? maybe moving it a bit further away from the cursor would already help?

**Review by abeforgit — COMMENTED** [review]
Don't really know how to solve this, but if you select multiple cells, the selection color overrides the background color, so it kinda seems like the cell color feature isn't doing anything (until you deselect)

If you know of a quick solution, it would be nice (maybe something with the alpha channel on css `rgba()` colors? or some opacity setting?)

**Review by abeforgit — APPROVED** [review]

**Pull request merged**